dpctl/tensor/_ctors.py

```diff
@@ -1035,3 +1035,82 @@ def linspace(
     )
     hev.wait()
     return res
+
+
+def eye(
+    n_rows,
+    n_cols=None,
+    /,
+    *,
+    k=0,
+    dtype=None,
+    order="C",
+    device=None,
+    usm_type="device",
+    sycl_queue=None,
+):
+    """
+    eye(n_rows, n_cols = None, /, *, k = 0, dtype = None, \
+    device = None, usm_type="device", sycl_queue=None) -> usm_ndarray
+
+    Creates `usm_ndarray` with ones on the `k`th diagonal.
+
+    Args:
+        n_rows: number of rows in the output array.
+        n_cols (optional): number of columns in the output array. If None,
+        n_cols = n_rows. Default: `None`.
+        k: index of the diagonal, with 0 as the main diagonal.
+        A positive value of k is a superdiagonal, a negative value
+        is a subdiagonal.
+        Raises `TypeError` if k is not an integer.
+        Default: `0`.
+        dtype (optional): data type of the array. Can be typestring,
+            a `numpy.dtype` object, `numpy` char string, or a numpy
+            scalar type. Default: None
+        order ("C" or F"): memory layout for the array. Default: "C"
+        device (optional): array API concept of device where the output array
+            is created. `device` can be `None`, a oneAPI filter selector string,
+            an instance of :class:`dpctl.SyclDevice` corresponding to a
+            non-partitioned SYCL device, an instance of
+            :class:`dpctl.SyclQueue`, or a `Device` object returnedby
+            `dpctl.tensor.usm_array.device`. Default: `None`.
+        usm_type ("device"|"shared"|"host", optional): The type of SYCL USM
+            allocation for the output array. Default: `"device"`.
+        sycl_queue (:class:`dpctl.SyclQueue`, optional): The SYCL queue to use
+            for output array allocation and copying. `sycl_queue` and `device`
+            are exclusive keywords, i.e. use one or another. If both are
+            specified, a `TypeError` is raised unless both imply the same
+            underlying SYCL queue to be used. If both are `None`, the
+            `dpctl.SyclQueue()` is used for allocation and copying.
+            Default: `None`.
+    """
+    if n_cols is None:
```

Comment:
Since order is assumed in the code to be either `"C"` or `"F"`, perhaps argument validation is in order.